### PR TITLE
`Development`: Use @EnforceAdmin annotation on class level

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).

### Motivation and Context

https://github.com/ls1intum/Artemis/pull/9296#pullrequestreview-2334081358: "EnforceAdmin should be on the class level"

### Description

We now use @EnforceAdmin on the class level for admin resources.
Additionally add two Arch rules: 
- Test the consistent usage of "Admin" in the name of the resource and the @EnforceAdmin annotation.
- Test that no other annotation on the method level overrides the @EnforceAdmin annotation

### Steps for Testing

Check that admin endpoints still work, e.g. use some functionality from the top right dropdown as admin locally or on TS3.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Manual Test 1
- [x] Manual Test 2

## Release Notes

- **New Features**
	- Centralized admin access control for multiple resource classes, ensuring all methods require admin privileges through class-level annotations.
	- Streamlined feature management with class-level feature toggle annotations for various resources.
	- Enhanced handling of feature toggles for improved flexibility and maintainability.

- **Bug Fixes**
	- Improved architectural testing rules related to the `@EnforceAdmin` annotation, ensuring proper naming conventions and preventing conflicting security annotations.

These changes streamline access control across the application, enhancing security and maintainability.

### CODE REVIEW

**Review by SimonEntholzer — COMMENTED** [review]
Code changes make sense.
What's the idea behind having architectureTest classes like this, basically three times, for service, resource and repository, when they basically all have an identical method?
```
package de.tum.cit.aet.artemis.atlas.architecture;

import de.tum.cit.aet.artemis.shared.architecture.module.AbstractModuleServiceArchitectureTest;

class AtlasServiceArchitectureTest extends AbstractModuleServiceArchitectureTest {

    public String getModulePackage() {
        return ARTEMIS_PACKAGE + ".atlas";
    }
}
```
And it seems there are still some tests failing

**Comment by JohannesStoehr** [comment]
> Code changes make sense. What's the idea behind having architectureTest classes like this, basically three times, for service, resource and repository, when they basically all have an identical method?

They come from this PR, which was only visible, because after it was merged I haven't yet merged the changes into this PR. Not the changes are gone in the changed file section.

> And it seems there are still some tests failing

Also fixed by merging develop again

File: src/main/java/de/tum/cit/aet/artemis/atlas/web/admin/AdminStandardizedCompetencyResource.java
Comment (pzdr7) [inline src/main/java/de/tum/cit/aet/artemis/atlas/web/admin/AdminStandardizedCompetencyResource.java:41]: Seems like this change is causing exceptions when the tests access the corresponding endpoints. Can you have a look?

**Review by pzdr7 — CHANGES_REQUESTED** [review]
Can you have a look at the server test failures related to standardized competencies?

**Review by az108 — DISMISSED** [review]
Tested locally, works as expected without any errors in admin functionalities

File: src/main/java/de/tum/cit/aet/artemis/atlas/web/admin/AdminStandardizedCompetencyResource.java
Comment (JohannesStoehr) [reply src/main/java/de/tum/cit/aet/artemis/atlas/web/admin/AdminStandardizedCompetencyResource.java:41]: There was a bug in the FeatureToggleAspect that I fixed now

**Review by az108 — DISMISSED** [review]
reapprove

File: src/main/java/de/tum/cit/aet/artemis/core/service/feature/FeatureToggleAspect.java
Comment (JohannesStoehr) [inline src/main/java/de/tum/cit/aet/artemis/core/service/feature/FeatureToggleAspect.java:34]: This is the same as for the „AtLeastTutor“ annotation and so on

**Review by pzdr7 — DISMISSED** [review]
Code 👍

**Review by az108 — APPROVED** [review]
reapprove

**Review by sarpsahinalp — APPROVED** [review]
Code lgtm!

**Review by iyannsch — DISMISSED** [review]
Code LGTM, locally all Admin features still work

**Review by az108 — DISMISSED** [review]

**Review by N0W0RK — DISMISSED** [review]
Tested locally. All endpoints are accessible as admin and not as user

**Review by MaximilianAnzinger — DISMISSED** [review]
Maintainer approved

**Review by Jan-Thurner — APPROVED** [review]
Code LGTM

**Review by iyannsch — APPROVED** [review]
Reapprove from stacked PR

**Review by pzdr7 — APPROVED** [review]
Reapprove after merge

**Review by az108 — APPROVED** [review]
reapprove merge

**Review by raffifasaro — APPROVED** [review]
Code LGMT

**Pull request merged**
